tests/test_optimization.py

```diff
@@ -116,3 +116,241 @@ def test_tantrix_area(self):
+        
+        # Verify basic properties
+        self.assertEqual(fitted_points.shape, (8, 3))
+        self.assertFalse(jnp.allclose(fitted_points, 0.0))
+        
+        # ← NUEVO: Verify convergence behavior
+        if len(loss_history) > 1:
+            # Check that loss generally decreases (allowing some fluctuation)
+            final_loss = loss_history[-1]
+            initial_loss = loss_history[0]
+            self.assertLess(final_loss, initial_loss * 1.5, 
+                        "Loss should decrease or at least not increase significantly")
+            
+            # Check that optimization is actually changing the loss
+            self.assertGreater(len(set(loss_history)), 1, 
+                            "Loss should change during optimization")
+
+    def test_error_handling_and_parameter_validation(self):
+        """
+        Test error handling, backward compatibility, and parameter validation.
+        Ensures the API works correctly and fails gracefully with invalid inputs.
+        """
+        # Test 1: Traditional random initialization should work unchanged
+        self.barqcurve.initialize_parameters(seed=42)
+        params_random = self.barqcurve.params['free_points']
+        
+        # Re-initialize with same seed should give same result
+        self.barqcurve.initialize_parameters(seed=42)
+        params_random_repeat = self.barqcurve.params['free_points']
+        self.assertTrue(jnp.allclose(params_random, params_random_repeat))
+        
+        # Test 2: Manual initialization should work unchanged
+        manual_points = jnp.ones((8, 3)) * 0.5
+        self.barqcurve.initialize_parameters(init_free_points=manual_points)
+        self.assertTrue(jnp.allclose(self.barqcurve.params['free_points'], manual_points))
+        
+        # Test 3: Error when conflicting initialization methods are used
+        with self.assertRaises(ValueError):
+            self.barqcurve.initialize_parameters(
+                init_free_points=manual_points,
+                fit_target_curve=lambda t: [t, t, t]
+            )
+        
+        # Test 4: Error when neither curve nor points provided to fitting
+        with self.assertRaises(ValueError):
+            self.barqcurve._fit_curve_to_control_points()
+        
+        # Test 5: Invalid curve function should be handled gracefully
+        def invalid_curve(t):
+            return [float('inf'), float('nan'), 0.0]
+        
+        # Should not crash, but may produce poor results
+        try:
+            self.barqcurve.initialize_parameters(fit_target_curve=invalid_curve)
+            # If it doesn't crash, verify the result is at least structurally valid
+            self.assertIsNotNone(self.barqcurve.params['free_points'])
+        except (ValueError, RuntimeError):
+            # It's acceptable for invalid inputs to raise errors
+            pass
+
+    def test_fitting_preserves_barq_structure(self):
```

Comment:
This test can be removed entirely because these constraints are handled internally by the barq functions, and thus out of scope for testing here.